### PR TITLE
TCP Upstream: Count bytes

### PR DESCRIPTION
**Pull request by KBaichoo**
Commit Message: Count  bytes passing through to TCP upstream.
Additional Description:
Risk Level: low
Testing: added integration test
Docs Changes: n/a
Release Notes: included
Platform Specific Features: n/a

### CODE REVIEW

File: test/config/utility.cc [hunk 1]
```diff
@@ -822,7 +822,8 @@ void ConfigHelper::addClusterFilterMetadata(absl::string_view metadata_yaml,
 
 void ConfigHelper::setConnectConfig(
     envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager& hcm,
-    bool terminate_connect, bool allow_post, bool http3) {
+    bool terminate_connect, bool allow_post, bool http3,
+    absl::optional<envoy::config::core::v3::ProxyProtocolConfig::Version> proxy_protocol_version) {
```
Comment (adisuissa) [inline test/config/utility.cc:826]: nit: OptRef

File: test/integration/tcp_tunneling_integration_test.cc [hunk 2]
```diff
@@ -91,6 +94,27 @@ class ConnectTerminationIntegrationTest : public HttpProtocolIntegrationTest {
       ASSERT_TRUE(response_->waitForEndStream());
       ASSERT_FALSE(response_->reset());
     }
+
+    const int expected_wire_bytes_sent = 9;
```
Comment (adisuissa) [inline test/integration/tcp_tunneling_integration_test.cc:98]: nit: add comments why/how these values were computed.

File: test/integration/tcp_tunneling_integration_test.cc [hunk 3]
```diff
@@ -165,6 +189,12 @@ TEST_P(ConnectTerminationIntegrationTest, DownstreamClose) {
   // Tear down by closing the client connection.
   codec_client_->close();
   ASSERT_TRUE(fake_raw_upstream_connection_->waitForHalfClose());
+  const int expected_wire_bytes_sent = 5;
```
Comment (adisuissa) [inline test/integration/tcp_tunneling_integration_test.cc:192]: nit: add comments why/how these values were computed. (here and below)

**Review by adisuissa — APPROVED** [review]
LGTM, some small nits.
Thanks!

**Comment by KBaichoo** [comment]
Thanks for the review @adisuissa  will address the nits in a follow up.

File: test/config/utility.cc (hunk 1 above)
Comment (KBaichoo) [reply test/config/utility.cc:826]: as discussed out of band ref would be larger than underlying enum int32_t

**Pull request merged**